src/controllers/auth.js

```diff
@@ -325,10 +325,69 @@ const refreshAccessToken = async (req, res) => {
   }
 };
 
+const changePassword = async (req, res) => {
+  const userInfo = req.user;
+  const { currentPassword, newPassword } = req.body.fields;
+
+  try {
+    // get the user from the Db
+    const users = rowService.get({
+      tableName: '_users',
+      whereString: 'WHERE id=?',
+      whereStringValues: [userInfo.userId],
+    });
+
+    if (users.length <= 0) {
+      return res.status(401).send({ message: 'User not found' });
+    }
+
+    const user = users[0];
+
+    // check if the users current password is valid
+    const isMatch = await comparePasswords(
+      currentPassword,
+      user.hashed_password,
+    );
+
+    if (!isMatch) {
+      return res.status(401).send({ message: 'Invalid current password' });
+    }
+
+    // check if the new password is strong
+    if (['Too weak', 'Weak'].includes(checkPasswordStrength(newPassword))) {
```

Comment:
Again here we can see that these two are repeated and it increases the chance of misspelling and ...